netutils/netinit/Kconfig

```diff
@@ -87,10 +87,38 @@ config NETINIT_MONITOR
 		required for network initialization are never released.
 
 if NETINIT_MONITOR
+config NETINIT_ESTABLISH_POLL_RATE
+	int "The poll rate in seconds, to check for link establishment."
+	default 2
+	depends on !ARCH_PHY_INTERRUPT && ARCH_PHY_POLLED
+	---help---
+		The network monitor will check the PHY link state every
+		NETINIT_ESTABLISH_POLL_RATE seconds for link establishment.
+
+config NETINIT_LOSS_POLL_RATE
+	int "The poll rate in seconds, to check for link loss."
+	default 3
+	depends on !ARCH_PHY_INTERRUPT && ARCH_PHY_POLLED
+	---help---
+		The network monitor will check the PHY link state every
+		NETINIT_LOSS_POLL_RATE seconds for link loss.
+
+config NETINIT_FALLBACK
+	int "The number of failed DHCP attempts to fall back to using static settings"
+	default 5
+	depends on !ARCH_PHY_INTERRUPT && ARCH_PHY_POLLED && FSUTILS_IPCFG
```

Comment:
Since DHCP isn't relate to the PHY operation, I don't think there are any reason to disable FALLBACK in this case. Let's desgin the thing right at the first place. BTW, the option should depend on DHCPC instead, add rename to NETINIT_DHCPC_FALLBACK.

Reply:
@xiaoxiang781216 without indications of state change either via interrupts of polling fallback will not work correctly.

Reply:
> BTW, the option should depend on DHCPC instead, add rename to NETINIT_DHCPC_FALLBACK.

Rename done. dependency on NETINIT_DHCPC added in 078d39e5e and force pushed.

Reply:
> @xiaoxiang781216 without indications of state change either via interrupts of polling fallback will not work correctly.

Could you point out where the dhcp fallback code depend on the poll? I can't find there is the coupling between the fallback and polling. I expect that the dependence is:
```
config NETINIT_FALLBACK
	int "The number of failed DHCP attempts to fall back to using static settings"
	default 5
	depends on (ARCH_PHY_INTERRUPT || ARCH_PHY_POLLED) && FSUTILS_IPCFG
```

Reply:
It is the times,

Reply:
Please either fix this issue or give a technical reason to not fix it. The time isn't a good reason to ignore the right concern.

Reply:
LOL By time I meat the time the task waits. 

Have you read the code?  Interrupt mode is just that: Edges are created on interrupts. In Interrupt mode the task waits "Forever".  In polled mode the task waits at one of the 2 POLL rates. This is dependent  on the state. Since Polling is a waste of CPU, it is only used it this mode. But without the IRQ line it is the only way to do this.  The time the task waits is the N * the POLL rate for the fall back. It will not work, of should I say it will be WAY to long in IRQ mode.

Reply:
The polling logic exist for a long time before ARCH_PHY_POLLED is added by you recently. The only real difference is that ARCH_PHY_POLLED need a short poll interval but ARCH_PHY_INTERRUPT need a long poll interval. So the simple change is let user config the interval through Kconfig.
But DHCP should happen only after the network establish the physical link. Why you think the network status polling ncopule with DHCP?

Reply:
Just like the HW had to have the IRQ line the HW has to also support link management while active. See the mailing list for Greg concern regarding this.   

!ARCH_PHY_INTERRUPT && ARCH_PHY_POLLED is that enable. 

I t would not be good to allow the user to set poll rates on HW that does not tolerate it,  This is all documented in the HW Kconfig.

Reply:
@davids5 I am talking about DHCP fallback here, it isn't relate to the polling, rihgt? It is very bad to combine the unrelated patch into one PR, so please move DHCP fallback to the separated PR. So we can disscuss the patch individually.

Reply:
No the polling adds the temporal component, 

There is a **great** possibility we are miss-communicating here (you and I have been is this sort of cycle before)

@xiaoxiang781216 At this point rather continue to say the same things over and over and over again. I invite you to do a PR against this branch with any changes you want. Verify it builds 100% with all the CONFIG* permutations. If it 1) works the same for me with, my setting AND 2) It does not expose the user to be able to configure the system in a way that will not work with their SoC's MAC and PHY, I will be happy to test it with my settings and merge your work into this PR.

Does that work for you?

Reply:
> No the polling adds the temporal component,
> 
> There is a **great** possibility we are miss-communicating here (you and I have been is this sort of cycle before)
> 

That is because you reject other people's comments, but don't give the technical reason.

> @xiaoxiang781216 At this point rather continue to say the same things over and over and over again. I invite you to do a PR against this branch with any changes you want. Verify it builds 100% with all the CONFIG* permutations. If it 1) works the same for me with, my setting AND 2) It does not expose the user to be able to configure the system in a way that will not work with their SoC's MAC and PHY, I will be happy to test it with my settings and merge your work into this PR.
> 
> Does that work for you?

Could you follow my suggestion to split the PR first?

Reply:
I am afraid I have run out of time. I need all the features herein to fix all the issues I have documented.